### PR TITLE
Tags instead of Categories

### PR DESCRIPTION
**Pull request by hkbertoson**
This is the start of working on #146 for using Tags instead of Categories.

### CODE REVIEW

File: src/components/Tags.astro [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	}
+	.tags-container {
+		display: flex;
+		flex-wrap: wrap;
+	}
+</style>
+
+<div class="tags-container">
+	{filteredTags.map((tag) => <span class="tag">{tag}</span>)}
+</div>
```
Comment (alexanderniebuhr) [inline src/components/Tags.astro:41]: Could we add a list of all resources to this page, which can be filtered if someone clicks on one of the tags from the list?

Comment (alexanderniebuhr) [reply src/components/Tags.astro:41]: After looking at this myself, it is not that easy.. Need to investigate further, will update later.

**Review by alexanderniebuhr — CHANGES_REQUESTED** [review]
Blocking, until we figure out how to implement filtering..

**Comment by BryceRussell** [comment]
Idea for filtering server-side that is compatible with Starlight/`hybrid` mode:

1. Create a SSR partial route with `export const prerender = false;` and `export const partial = true`
2. Author partial route to render a filtered version of the list using data from the request (JSON, query parameters, form data, etc)
3. Filter the list by fetching the partial client side and swapping the old list with the new filtered list

**Comment by hkbertoson** [comment]
I think query parameters would be the best option. Then that would allow users to share a URL with the filters.

**Comment by alexanderniebuhr** [comment]
> I think query parameters would be the best option. Then that would allow users to share a URL with the filters.

We looked at that during the call, it doesn't work, since Starlight is prerendered. And I don't want to do some workarounds with inline scripts. But as @BryceRussell we could use some "island" partial solution. 

Another idea would be to always inline all items inside the HTML, and only do client-side filtering, should be fast enough for a long time too.

**Comment by OliverSpeir** [comment]
@HiDeoo Any suggestions on this?

**Comment by alexanderniebuhr** [comment]
> _https://github.com/astrolicious/astro-tips.dev/pull/155#issuecomment-2177454335_: @HiDeoo Any suggestions on this?

It was answered on Discord, which can be found [here](https://discord.com/channels/1217527207467946087/1217544238397526016/1252894745991315507).

File: astro.config.ts [hunk 2]
```diff
@@ -57,9 +57,10 @@ export default defineConfig({
 				{
 					label: 'Resources',
 					badge: 'New',
-					autogenerate: {
-						directory: 'resources',
-					},
+					items: [
+						{ slug: 'resources/educational' },
```
Comment (florian-lefebvre) [inline astro.config.ts:61]: Do we need this anymore? I'd say tags should replace the current system

File: src/content/config.ts [hunk 3]
```diff
@@ -45,6 +45,7 @@ const resourcesSchema = defineCollection({
 			'3rd-party',
 			'migration',
 		]),
+		tags: z.array(z.string()),
```
Comment (florian-lefebvre) [inline src/content/config.ts:48]: Maybe we want to use a reference to a new `tags` collection to have proper display names?

File: src/content/resources/astro-db-migrating-analytics-postgres.json
Comment (florian-lefebvre) [inline src/content/resources/astro-db-migrating-analytics-postgres.json:1]: let's get rid of the category then?

File: src/content/docs/resources/tags.mdx
Comment (florian-lefebvre) [inline src/content/docs/resources/tags.mdx:1]: I think this page should be removed?

**Review by alexanderniebuhr — APPROVED** [review]

File: src/content/config.ts (hunk 3 above)
Comment (florian-lefebvre) [reply src/content/config.ts:31]: in a follow up pr

**Pull request merged**